### PR TITLE
Fix/update script for bumblebee activation.

### PR DESCRIPTION
**Pull request by deiferni**
Fix existing reindex script to actually commit the changes (Follow-up of https://github.com/4teamwork/opengever.core/pull/1980).

Add feature to calculate bumblebee checksums for archived documents:

Adds an `Archeologist` which can excavate old archived versions of objects. It returns a direct reference to the archived object by bypassing all the other layers of abstraction. For more details please see `Archeologist`'s docstring.

The script was also tested manually by:
- creating a random amount of versions for ~1000 documents on our dev-server
- generating checksums for these versions
- validating that checksums were present for all versions

### CODE REVIEW

**Comment by deiferni** [comment]
@phgross please have a thorough look

File: opengever/base/archeologist.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
+from plone import api
+from Products.CMFEditions.utilities import dereference
+
+
+class Archeologist(object):
+    """Dig out an archived version of an object.
```
Comment (lukasgraf) [inline opengever/base/archeologist.py]: dig up

File: scripts/bumblebee_installation.py [hunk 2]
```diff
@@ -5,35 +5,48 @@
...
 from optparse import OptionParser
+from plone import api
+from zope.annotation.interfaces import IAnnotations
 from zope.component import getUtility
 import logging
 import sys
+import transaction
+
 
 # Set global logger to info - this is necessary for the log-outbut with
```
Comment (lukasgraf) [inline scripts/bumblebee_installation.py]: `log-output`

File: opengever/base/archeologist.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+    Parameters:
+    obj - the current plone object for which you want to access the archived
+          object
+    version_data - a VersionData instance as returned by `portal_repository`
+
+    The layers of abstraction that are involded (in order of appearance):
```
Comment (lukasgraf) [inline opengever/base/archeologist.py]: `involved`

File: opengever/base/archeologist.py (hunk 1 above)
Comment (lukasgraf) [inline opengever/base/archeologist.py]: Maybe something like "a mutable, archived version of an object. BlobFiles not included.".

File: opengever/base/archeologist.py [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+        self.obj, self.history_id = dereference(obj)
+        self.selector = version_data.version_id
+        self.version_data = version_data
+        self.storage = api.portal.get_tool('portal_historiesstorage')
+
+    def excavate(self):
+        """Return a reference to the archived object.
+
+        Bypass all the copies made by the different layers of abstraction and
+        dig out a reference to the archived object.
```
Comment (lukasgraf) [inline opengever/base/archeologist.py]: dig up

File: scripts/bumblebee_installation.py [hunk 5]
```diff
@@ -89,7 +105,34 @@ def main(app, argv=sys.argv[1:]):
 
     if mode == 'reindex':
         LOG.info("Start indexing objects...")
-        return converter.reindex()
+        converter.reindex()
+        return transaction.commit()
+
+    elif mode == 'history':
+        LOG.info("Start creating cehcksums for portal repository ...")
```
Comment (lukasgraf) [inline scripts/bumblebee_installation.py]: czechsums? 😉

**Comment by lukasgraf** [comment]
Other than that: 👍 

💚  the docs and the name 😉

**Pull request merged**